tests/test_facets_control_safety.py

```
@@ -0,0 +1,340 @@
+        self.calls += 1
+        raise AssertionError("caller callback executed")
+
+
+def _hostile_int(counter: _CallbackCounter) -> int:
+    """Return an int subclass whose conversion/comparison/hash hooks fail."""
+
+    class HostileInt(int):
+        def __int__(self):
+            counter.hit()
+
+        def __index__(self):
```

Ruff (0.16.1)

[error] `__index__` does not return an integer

(PLE0305)